engine/src/main/java/org/hibernate/validator/internal/properties/DefaultGetterPropertyMatcher.java

```diff
@@ -47,52 +41,38 @@
 	 * 		{@code false} otherwise.
 	 */
 	@Override
-	public boolean isProperty(Executable executable) {
-		if ( executable instanceof Constructor ) {
-			return false;
-		}
+	public boolean isProperty(ConstrainableExecutable executable) {
 		if ( executable.getParameterTypes().length != 0 ) {
 			return false;
 		}
-		Method method = (Method) executable;
 
-		String methodName = method.getName();
+		String methodName = executable.getName();
 
 		//<PropertyType> get<PropertyName>()
-		if ( methodName.startsWith( PROPERTY_ACCESSOR_PREFIX_GET ) && method.getReturnType() != void.class ) {
+		if ( methodName.startsWith( PROPERTY_ACCESSOR_PREFIX_GET ) && executable.getReturnType() != void.class ) {
 			return true;
 		}
 		//boolean is<PropertyName>()
-		else if ( methodName.startsWith( PROPERTY_ACCESSOR_PREFIX_IS ) && method.getReturnType() == boolean.class ) {
+		else if ( methodName.startsWith( PROPERTY_ACCESSOR_PREFIX_IS ) && executable.getReturnType() == boolean.class ) {
 			return true;
 		}
 		//boolean has<PropertyName>()
-		else if ( methodName.startsWith( PROPERTY_ACCESSOR_PREFIX_HAS ) && method.getReturnType() == boolean.class ) {
+		else if ( methodName.startsWith( PROPERTY_ACCESSOR_PREFIX_HAS ) && executable.getReturnType() == boolean.class ) {
 			return true;
 		}
 
 		return false;
 	}
 
 	@Override
-	public String getPropertyName(Method method) {
+	public String getPropertyName(ConstrainableExecutable executable) {
 		String name = null;
-		String methodName = method.getName();
+		String methodName = executable.getName();
 		for ( String prefix : PROPERTY_ACCESSOR_PREFIXES ) {
 			if ( methodName.startsWith( prefix ) ) {
 				name = StringHelper.decapitalize( methodName.substring( prefix.length() ) );
```

Comment:
You should `break;` from here. Otherwise if you have an `isgethas()`, you will have some issues (agreed it's not recommended but let's be thorough).